### PR TITLE
NET-1226: Scalability Improvements

### PR DESCRIPTION
**Pull request by yabinma**
## Checklist before requesting a review
- [ ] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netclient & Netmaker are awesome.

### CODE REVIEW

File: functions/mqpublish.go [hunk 1]
```diff
@@ -41,21 +41,61 @@ func Checkin(ctx context.Context, wg *sync.WaitGroup) {
...
+			if config.CurrServer == "" {
+				continue
+			}
+			server := config.GetServer(config.CurrServer)
+			if server == nil {
+				slog.Warn("server config is nil")
+				continue
+			}
+			if Mqclient == nil || !Mqclient.IsConnectionOpen() {
+				slog.Warn("MQ client is not connected, using fallback checkin for server", config.CurrServer)
```
Comment (abhishek9686) [inline functions/mqpublish.go:63]: if mqclient is not connected, publish metrics should be called with fallback

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: functions/mqpublish.go (hunk 1 above)
Comment (yabinma) [reply functions/mqpublish.go:58]: added fallback, review again please.

**Review by abhishek9686 — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM gravitl/go-builder as builder
+FROM gravitl/go-builder AS builder
```
Comment (abhishek9686) [inline Dockerfile:1]: is this change intended?

Comment (yabinma) [reply Dockerfile:1]: Yes, in the new docker version,   the syntax  check becomes more strict.   "FROM,  as"  receives warning message.   "FROM,  AS" works with both of UPPERCASE.

File: functions/mqpublish.go
Comment (abhishek9686) [inline functions/mqpublish.go:1]: can we cleanup the commented code...
and rename the fallback func `hostUpdateFallback` -> `hostServerUpdate`

File: functions/pull.go [hunk 3]
```diff
@@ -79,6 +84,8 @@ func Pull(restart bool) (models.HostPull, bool, bool, error) {
 	if restart {
 		logger.Log(3, "restarting daemon")
 		return models.HostPull{}, resetInterface, replacePeers, daemon.Restart()
+	} else {
+		_ = wireguard.SetPeers(replacePeers)
```
Comment (abhishek9686) [inline functions/pull.go:88]: why are we setting peers here?

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: functions/mqpublish.go
Comment (yabinma) [reply functions/mqpublish.go:1]: Done.  Added a commit for changing the name.

File: functions/pull.go (hunk 3 above)
Comment (yabinma) [reply functions/pull.go:88]: Without it, some of the new addded peers will be missed in the wg peers as tested.

Comment (abhishek9686) [reply functions/pull.go:88]: this will be a duplicate call, we already set peers outside this func

Comment (yabinma) [reply functions/pull.go]: commit added.
